enterprise/backend/src/metabase_enterprise/api/routes/common.clj

```diff
@@ -16,9 +16,11 @@
   if we do not."
   [feature feature-name handler]
   (assert (i18n/localized-string? feature-name), "`feature-name` must be i18ned")
-  (fn [request respond raise]
-    (premium-features/assert-has-feature feature feature-name)
-    (handler request respond raise)))
+  (with-meta
+   (fn [request respond raise]
```

Comment:
I saw a weird thread on clojurians slack about functions not "technically" having support for metadata, and that it's "an implementation detail". I think this is fine though, I doubt they'll ever change it. [thread here](https://clojurians.slack.com/archives/C06E3HYPR/p1709908290330649)

Reply:
Uhhh weird right?! Anyway, I think we don't have too many options here - operating vars in this place would _weird_.